### ISSUE TITLE
Typescript prop validation from `.attrs`

### ISSUE THREAD

**Issue by matthova** [styled-system/styled-system#1638]
Not sure if I'm using styled-system incorrectly or if this isn't possible, but I'm currently unable to infer the theme's props when calling styled-system props from the `.attrs` function

I've constructed an example in code-sandbox [here](https://codesandbox.io/s/flamboyant-solomon-16njr?file=/src/App.tsx)

```tsx
import { color, ColorProps } from "styled-system";
import { styled } from "./styled-components";
import { theme } from "./theme";

const Box = styled("div")<ColorProps<typeof theme>>`
  ${color}
`;

// theme's colors are "primary" and "secondary"
// Is there a way to ensure type-safety when declaring color
// from the ".attrs" function?
const PrimaryBox = styled(Box).attrs({ backgroundColor: "primary" })``;

// This should throw a ts error since backgroundColor is not a key of theme colors
const TypescriptErrorBox = styled(Box).attrs({ backgroundColor: "primaryy" })``;
```

**Comment by matthova** [comment]
I'm wondering if this is more an issue with styled-components than styled-system, but in all cases, I'd love to inject some knowledge into `.attrs`.
